### PR TITLE
scan: flow.cylc file compatibility

### PR DESCRIPTION
**Pull request by oliver-sanders**
Fixes the integration test battery.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.
- [x] No dependency changes.

### CODE REVIEW

**Review by kinow — APPROVED** [review]
:+1: although it's weird that only `FLOW_FILE` is accessed directly. Should we—later—drop the other `Path`'s? Or maybe use `or FLOW_FILE.name in listing`?

File: cylc/flow/network/scan.py [hunk 1]
```diff
@@ -73,8 +73,6 @@
 
 SERVICE = Path(SuiteFiles.Service.DIRNAME)
 CONTACT = Path(SuiteFiles.Service.CONTACT)
```
Comment (oliver-sanders) [inline cylc/flow/network/scan.py:75]: Used in the scan function for path wrangling.

File: cylc/flow/network/scan.py [hunk 2]
```diff
@@ -73,8 +73,6 @@
...
-SUITERC = Path(SuiteFiles.SUITE_RC)
-FLOW_FILE = Path(SuiteFiles.FLOW_FILE)
```
Comment (oliver-sanders) [inline cylc/flow/network/scan.py:77]: Not/no longer used elsewhere.

File: cylc/flow/network/scan.py [hunk 3]
```diff
@@ -94,9 +92,9 @@ def dir_is_flow(listing):
         for path in listing
     }
     return (
-        SERVICE.name in listing
-        or SUITERC.name in listing  # cylc7 flow definition file name
-        or FLOW_FILE in listing  # cylc8 flow definition file name
+        SuiteFiles.SERVICE in listing
```
Comment (hjoliver) [inline cylc/flow/network/scan.py:95]:
```suggestion
        SERVICE in listing
```
Causing tests to fail.

**Review by kinow — APPROVED** [review]
CI passing, code looks good (neat trick with `bool(set & set)`! Gotta remember that)

**Pull request merged**